### PR TITLE
Skip checking installed packages in mode test

### PR DESCRIPTION
**Pull request by teclator**
- https://bugzilla.suse.com/show_bug.cgi?id=1138668

## Problem
- Failing old testsuite in yast2-dns-server
- It depends on bind rpm installed.

## Fix

- Do not depend on the package installed at all in Mode.test()
- 4.2.1

### CODE REVIEW

File: testsuite/tests/DnsServerAPI.rb [hunk 1]
```diff
@@ -84,32 +84,6 @@ def main
           }
         },
         "sysconfig" => {
-          "SuSEfirewall2"     => {
```
Comment (imobachgs) [inline testsuite/tests/DnsServerAPI.rb]: I guess it is an unrelated clean-up, right?

File: src/modules/DnsServer.pm [hunk 2]
```diff
@@ -1309,6 +1309,7 @@ sub update_forwarding {
 # when applying configuration in AutoYast config module
 sub check_and_install_package {
     my $self = shift;
+    return 1 if (Mode->test ());
```
Comment (imobachgs) [inline src/modules/DnsServer.pm]: Same as https://github.com/yast/yast-yast2/pull/947#pullrequestreview-256409342.

File: testsuite/tests/DnsServerAPI.rb (hunk 1 above)
Comment (teclator) [reply testsuite/tests/DnsServerAPI.rb]: Unrelated, yes, but as SuSEfirewall2 is now unsupported this mockup is invalid or not necessary.
